spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/MapTaskExecutionDao.java

```diff
@@ -46,7 +51,67 @@ public void updateTaskExecution(TaskExecution taskExecution) {
 		taskExecutions.put(taskExecution.getExecutionId(), taskExecution);
 	}
 
-	public Map<String, TaskExecution> getTaskExecutions(){
+	@Override
+	public TaskExecution getTaskExecution(String executionId) {
+		return taskExecutions.get(executionId);
+	}
+
+	@Override
+	public long getTaskExecutionCount(String taskName) {
+		int count = 0;
+		for (Map.Entry<String, TaskExecution> entry : taskExecutions.entrySet()) {
+			if (entry.getValue().getTaskName().equals(taskName)) {
+				count++;
+			}
+		}
+		return count;
+	}
+
+	@Override
+	public Set<TaskExecution> findRunningTaskExecutions(String taskName) {
+		Set<TaskExecution> result = new HashSet<>();
+		for (Map.Entry<String, TaskExecution> entry : taskExecutions.entrySet()) {
+			if (entry.getValue().getTaskName().equals(taskName) &&
+					entry.getValue().getEndTime() == null) {
+				result.add(entry.getValue());
+			}
+		}
+		return Collections.unmodifiableSet(result);
+	}
+
+	@Override
+	public List<TaskExecution> getTaskExecutionsByName(String taskName, int start, int count) {
+		List<TaskExecution> result = new ArrayList<>();
+		Set<TaskExecution> filteredSet = new HashSet<>();
+		for (Map.Entry<String, TaskExecution> entry : taskExecutions.entrySet()) {
+			if (entry.getValue().getTaskName().equals(taskName)) {
+				filteredSet.add(entry.getValue());
+			}
+		}
+		int rowNum = 0;
+		Iterator<TaskExecution> rs = filteredSet.iterator();
+		while (rowNum < start && rs.hasNext()) {
+			rs.next();
+			rowNum++;
+		}
+		while (rowNum < start + count && rs.hasNext()) {
+			result.add(rs.next());
+			rowNum++;
+		}
+
+		return Collections.unmodifiableList(result);
```

Comment:
Do we need to wrap this given the fact that it's a new data structure for each call?